src/ml/neural_net/mps_compute_context.mm

```diff
@@ -16,6 +16,19 @@
 namespace turi {
 namespace neural_net {
 
+namespace {
+
+std::unique_ptr<compute_context> create_mps_compute_context() {
+  return std::unique_ptr<compute_context>(new mps_compute_context);
+}
+
+// At static-init time, register create_mps_compute_context().
+// TODO: Codify priority levels?
+static auto* mps_registration = new compute_context::registration(
```

Comment:
Note that with this chance, `unity_neural_net` no longer even links against `tcmps`... The only reason there's still a dependency in the CMakeLists is as a convenient way to force `tcmps` to be built and linked in the final binary

For a Python-based library, we can just have that library sit on top (not be a dependency of unity_shared), and just inject itself at run time in the same way as this code does